### PR TITLE
Add unit tests for github paginated request functions

### PR DESCRIPTION
**Pull request by shay2025**
## Description

This pull request introduces unit tests for github paginated request related functions.

## Related issue

*None*

## Type of change

Improvements (non-breaking change without functionality)

## How was this tested?

Ran `task test` command.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have ran `task check -f` and have no issues

### CODE REVIEW

**Comment by reviewpad-team** [comment]
**Reviewpad Report**

**:bangbang: Errors**
* **Unconventional commit detected**: 'Add unit tests for github paginated request functions' (678b33b5ed1fd3dae7d525568804648622a36ca3)

File: utils/github_test.go [hunk 1]
```diff
@@ -5,14 +5,21 @@
...
+	"net/http"
 	"testing"
 
+	"github.com/google/go-github/v42/github"
 	mocks_aladino "github.com/reviewpad/reviewpad/v2/mocks/aladino"
 	"github.com/reviewpad/reviewpad/v2/utils"
 	"github.com/stretchr/testify/assert"
 )
 
+type result struct {
```
Comment (ferreiratiago) [inline utils/github_test.go:19]: `result` is too generic. What is is this `result`?

File: utils/github_test.go [hunk 2]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+	}
+
+	wantRes := []*result{{pageNum: 1}}
+	gotRes, err := utils.PaginatedRequest(initFn, reqFn)
+
+	assert.Nil(t, err)
+	assert.Equal(t, gotRes, wantRes)
+}
+
+func TestParseNumPagesFromLink_WhenURLHasNoInfo(t *testing.T) {
```
Comment (ferreiratiago) [inline utils/github_test.go:137]: Suggestion `TestParseNumPagesFromLink_WhenHTTPLinkHeaderHasNoRel`

File: utils/github_test.go [hunk 3]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+	link := "<https://api.github.com/user/58276/repos?page=1>"
+
+	wantNumPages := 0
+
+	gotNumPages := utils.ParseNumPagesFromLink(link)
+
+	assert.Equal(t, wantNumPages, gotNumPages)
+}
+
+func TestParseNumPagesFromLink_WhenParseURLFails(t *testing.T) {
```
Comment (ferreiratiago) [inline utils/github_test.go:147]: Suggestion `TestParseNumPagesFromLink_WhenHTTPLinkHeaderIsInvalid`

File: utils/github_test.go [hunk 4]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+	link := "<invalid%+url>; rel=\"last\""
+
+	wantNumPages := 0
+
+	gotNumPages := utils.ParseNumPagesFromLink(link)
+
+	assert.Equal(t, wantNumPages, gotNumPages)
+}
+
+func TestParseNumPagesFromLink_WhenPageQueryIsNotPresentInURL(t *testing.T) {
```
Comment (ferreiratiago) [inline utils/github_test.go:157]: Suggestion `TestParseNumPagesFromLink_WhenHTTPLinkHeaderHasNoQueryParamPage`

File: utils/github_test.go [hunk 5]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+	link := "<https://api.github.com/user/58276/repos>; rel=\"last\""
+
+	wantNumPages := 0
+
+	gotNumPages := utils.ParseNumPagesFromLink(link)
+
+	assert.Equal(t, wantNumPages, gotNumPages)
+}
+
+func TestParseNumPagesFromLink_WhenNumPagesParseIntFails(t *testing.T) {
```
Comment (ferreiratiago) [inline utils/github_test.go:167]: Suggestion `TestParseNumPagesFromLink_WhenHTTPLinkHeaderHasInvalidQueryParamPage`

File: utils/github_test.go [hunk 6]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+
+	// The number of pages are given in the url query "page"
+	wantNumPages := 3
+
+	gotNumPages := utils.ParseNumPagesFromLink(link)
+
+	assert.Equal(t, wantNumPages, gotNumPages)
+}
+
+func TestParseNumPages_WhenNoLinkIsProvided(t *testing.T) {
```
Comment (ferreiratiago) [inline utils/github_test.go:188]: Suggestion `TestParseNumPages_WhenHTTPLinkHeaderIsNotProvided `

File: utils/github_test.go [hunk 7]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+
+	gotNumPages := utils.ParseNumPagesFromLink(link)
+
+	assert.Equal(t, wantNumPages, gotNumPages)
+}
+
+func TestParseNumPagesFromLink(t *testing.T) {
+	link := "<https://api.github.com/user/58276/repos?page=3>; rel=\"last\""
+
+	// The number of pages are given in the url query "page"
```
Comment (ferreiratiago) [inline utils/github_test.go:180]: Suggestion `// the number of pages is provided in the url query parameter "page"`

File: utils/github_test.go [hunk 8]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+func TestParseNumPages_WhenNoLinkIsProvided(t *testing.T) {
+	respHeader := make(http.Header)
+	respHeader.Add("Link", " ")
+	resp := &github.Response{
+		Response: &http.Response{
+			Header: respHeader,
+		},
+	}
+
+	// The number of pages are given in the url query "page"
```
Comment (ferreiratiago) [inline utils/github_test.go:197]: This comment is not necessary.

File: utils/github_test.go [hunk 9]
```diff
@@ -51,3 +58,163 @@ func TestGetPullRequestNumber(t *testing.T) {
...
+func TestParseNumPages(t *testing.T) {
+	respHeader := make(http.Header)
+	respHeader.Add("Link", "<https://api.github.com/user/58276/repos?page=3>; rel=\"last\"")
+	resp := &github.Response{
+		Response: &http.Response{
+			Header: respHeader,
+		},
+	}
+
+	// The number of pages are given in the url query "page"
```
Comment (ferreiratiago) [inline utils/github_test.go:214]: Suggestion `// the number of pages is provided in the url query parameter "page"`

**Review by ferreiratiago — CHANGES_REQUESTED** [review]
Please have a look at the comments.

File: utils/github_test.go (hunk 1 above)
Comment (shay2025) [reply utils/github_test.go:19]: `result` is an object that results of the paginated request

**Review by ferreiratiago — APPROVED** [review]

**Pull request merged**